src/contracts/pods/EigenPod.sol

```diff
@@ -59,43 +55,21 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
     /// @notice The single EigenPodManager for EigenLayer
     IEigenPodManager public immutable eigenPodManager;
 
-    ///@notice The maximum amount of ETH, in gwei, a validator can have restaked in the eigenlayer
-    uint64 public immutable MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR;
-
     /// @notice This is the genesis time of the beacon state, to help us calculate conversions between slot and timestamp
     uint64 public immutable GENESIS_TIME;
 
-    // STORAGE VARIABLES
-    /// @notice The owner of this EigenPod
-    address public podOwner;
-
-    /**
-     * @notice The latest timestamp at which the pod owner withdrew the balance of the pod, via calling `withdrawBeforeRestaking`.
-     * @dev This variable is only updated when the `withdrawBeforeRestaking` function is called, which can only occur before `hasRestaked` is set to true for this pod.
-     * Proofs for this pod are only valid against Beacon Chain state roots corresponding to timestamps after the stored `mostRecentWithdrawalTimestamp`.
-     */
-    uint64 public mostRecentWithdrawalTimestamp;
-
-    /// @notice the amount of execution layer ETH in this contract that is staked in EigenLayer (i.e. withdrawn from the Beacon Chain but not from EigenLayer),
-    uint64 public withdrawableRestakedExecutionLayerGwei;
-
-    /// @notice an indicator of whether or not the podOwner has ever "fully restaked" by successfully calling `verifyCorrectWithdrawalCredentials`.
-    bool public hasRestaked;
-
-    /// @notice This is a mapping of validatorPubkeyHash to timestamp to whether or not they have proven a withdrawal for that timestamp
-    mapping(bytes32 => mapping(uint64 => bool)) public provenWithdrawal;
+    /// @notice If a validator is slashed on the beacon chain and their balance has not been checkpointed
+    /// within `TIME_TILL_STALE_BALANCE` of the current block, they are eligible to be marked "stale"
+    /// via `verifyStaleBalance`.
+    uint256 internal constant TIME_TILL_STALE_BALANCE = 2 weeks;
 
-    /// @notice This is a mapping that tracks a validator's information by their pubkey hash
-    mapping(bytes32 => ValidatorInfo) internal _validatorPubkeyHashToInfo;
+    /// @notice A `verifyStaleBalance` proof MUST be older than `STALENESS_GRACE_PERIOD`, to give time
+    /// to a pod owner to prove the slashed validator's balance
+    uint256 internal constant STALENESS_GRACE_PERIOD = 6 hours;
```

Comment:
We may not need the grace period though, because the only repercussion is "start a checkpoint". In that case, the `TIME_TILL_STALE_BALANCE` might be sufficient.

After all, in order to do a proof, you'd need to do a checkpoint anyway :D

Reply:
(the repercussion used to be "worse", hence the grace period)

Reply:
[eec67ae](https://github.com/Layr-Labs/eigenlayer-contracts/pull/515/commits/eec67aeadf696f2058ff85e578531ee0dd0e20d5)